lib/src/signed_video_h26x_auth.c

```diff
@@ -152,19 +152,18 @@ verify_gop_hash(signed_video_t *self)
  * linked hash with the first NALU hash and marks it as used.
  */
 void
-update_link_hash_for_auth(signed_video_t *self, h26x_nalu_list_t *nalu_list)
+update_link_hash_for_auth(signed_video_t *self)
 {
   const size_t hash_size = self->verify_data->hash_size;
+  h26x_nalu_list_t *nalu_list = self->nalu_list;
```

Comment:
You only need `nalu_list` on the next line. How about `h26x_nalu_list_item_t *item = self->nalu_list->first_item;`